### PR TITLE
For issue #963 (dangers of SMS as a second factor)

### PR DESCRIPTION
**Pull request by NamiShuri**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [x] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [x] Any references to website have been formatted as [TEXT](URL “NAME”)
- [x] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR covers issue #963.

### CODE REVIEW

**Comment by commjoen** [comment]
Thank you for your PR! Looks good! Both Jeroens will review it soon ;-)

File: Document/0x04e-Testing-Authentication-and-Session-Management.md [hunk 1]
```diff
@@ -56,6 +56,25 @@ Two-factor authentication (2FA) is standard for apps that allow users to access
...
 
+Dangers of SMS-OTP
+
+ Although one-time passwords (OTP) sent via SMS are a common second factor for 2-factor authentication, this method has its shortcomings. In 2016, NIST suggested that "Due to the risk that SMS messages may be intercepted or redirected, implementers of new systems SHOULD carefully consider alternative authenticators." Below you will find a list of some related threats and suggestions to avoid successful attacks on SMS-OTP.  
+
+Threats:  
+
+-  Wireless Interception: The adversary can incept SMS messages by abusing femtocells and other known vulnerabilities in the SS7 network.  
+- Trojans: Malware downloaded on a device used to forward data sent in SMS messages to another number.  
+- SIM SWAP Attack: In this attack, the adversary calls the phone company and requests that a number be used with different sim card owned by the adversary. If successful, the adversary can see messages, and possibly use any 2-factor authentication that is sent to access accounts associated with that number.
```
Comment (commjoen) [inline Document/0x04e-Testing-Authentication-and-Session-Management.md]: Maybe we can rewrite the last sentence into: "If successful, the adversary can see the SMS messages which are send to the phone number. This includes the messages used in the 2-factor authentication."

Comment (TheDauntless) [reply Document/0x04e-Testing-Authentication-and-Session-Management.md]: For SIM Swap, I would add malicious insider at telecom center.

>- Trojans: Malware downloaded on a device used to forward data sent in SMS messages to another number.
To
> - Trojans: Installed malicious applications with access to text messages may forward the OTP to another number or backend.

> Wireless Interception: The adversary can incept SMS messages by abusing femtocells and other known vulnerabilities in the SS7 network.
  
To

> Wireless Interception: The adversary can intercept SMS messages by abusing femtocells and other known vulnerabilities in the SS7 network.

Comment (commjoen) [reply Document/0x04e-Testing-Authentication-and-Session-Management.md]: Another thing; maybe the SS7 is not the only network that might be problematic? (Given that there might be insecurities in 2G/3G and 4G as well for instance)

**Comment by NamiShuri** [comment]
WIll make the requested changes now

File: Document/0x04e-Testing-Authentication-and-Session-Management.md (hunk 1 above)
Comment (NamiShuri) [reply Document/0x04e-Testing-Authentication-and-Session-Management.md]: Just commenting to say I have read the comments, agree, and will make the necessary changes.

**Comment by commjoen** [comment]
Thank you for your contribution and fixing the feedback!

File: Document/0x04e-Testing-Authentication-and-Session-Management.md (hunk 1 above)
Comment (Akunwataa) [reply Document/0x04e-Testing-Authentication-and-Session-Management.md]: Please I thoroughly needs this ss7 software application to bypass Nigeria bank OTP verification of an account to assure a successful withdraw of cash

**Pull request merged**